openedx/core/djangoapps/schedules/template_context.py

```diff
@@ -1,36 +1,39 @@
-from urlparse import urlparse
+from urlparse import urlparse, parse_qs
 
+import attr
 from django.conf import settings
 from django.core.urlresolvers import reverse
-from django.utils.http import urlquote
+from django.utils.http import urlquote, urlencode
 
 from edxmako.shortcuts import marketing_link
+from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+from openedx.core.djangoapps.theming.helpers import get_current_site
 
 
-def get_base_template_context(site):
+def get_base_template_context(site, campaign=None):
     """Dict with entries needed for all templates that use the base template"""
     return {
         # Platform information
-        'homepage_url': encode_url(marketing_link('ROOT')),
-        'dashboard_url': absolute_url(site, reverse('dashboard')),
+        'homepage_url': encode_url(marketing_link('ROOT'), campaign=campaign),
+        'dashboard_url': absolute_url(site, reverse('dashboard'), campaign=campaign),
         'template_revision': settings.EDX_PLATFORM_REVISION,
         'platform_name': settings.PLATFORM_NAME,
         'contact_mailing_address': settings.CONTACT_MAILING_ADDRESS,
-        'social_media_urls': encode_urls_in_dict(getattr(settings, 'SOCIAL_MEDIA_FOOTER_URLS', {})),
-        'mobile_store_urls': encode_urls_in_dict(getattr(settings, 'MOBILE_STORE_URLS', {})),
+        'social_media_urls': getattr(settings, 'SOCIAL_MEDIA_FOOTER_URLS', {}),
+        'mobile_store_urls': getattr(settings, 'MOBILE_STORE_URLS', {}),
     }
 
 
-def encode_url(url):
-    # Sailthru has a bug where URLs that contain "+" characters in their path components are misinterpreted
-    # when GA instrumentation is enabled. We need to percent-encode the path segments of all URLs that are
-    # injected into our templates to work around this issue.
+def encode_url(url, campaign=None):
     parsed_url = urlparse(url)
-    modified_url = parsed_url._replace(path=urlquote(parsed_url.path))
+    parsed_qs = parse_qs(parsed_url.query)
+    if campaign is None:
+        campaign = CampaignTrackingInfo()
+    modified_url = parsed_url._replace(query=campaign.to_query_string(parsed_qs))
```

Comment:
How come we are not `urlqoute`ing the path anymore?

Since this function does not do that anymore, shouldn't it be named `add_campaign_utm_to_url` or something similar?

Reply:
That urlquote thing was a hack to work around a bug in Sailthru's link re-writing logic. Their support people said that if we turned off GA integration then we wouldn't need to do that anymore.

The plan is to disable their GA integration since we are handling it ourselves now.